### PR TITLE
Update game stat tests

### PR DESCRIPTION
**Pull request by ryandbarnett**
I think I was adding to the confusion around mocking and stubbing. I'm going to put this down as one of the challenges of this project. **I still need to learn more about mocks.**

This PR changes the following:

- ~~Removes use of Struct in favor of stub~~ Removes stub in favor of mock!
- Removes any use of "mock" in variable names to avoid confusion
- Leaves a comment at the top telling devs what datasets and attributes are used in this test file

### CODE REVIEW

**Comment by RyanDBarnett** [comment]
Here are the explanation from the article I referenced above that helped me.

**Stub explanation**

_A Test Stub is a fake object that’s used in place of a real object for the purpose of getting the program to behave the way we need it to in order to test it. A big part of a Test Stub’s job is to return pre-specified hard-coded values in response to method calls._

**Mock explanation**

_A Mock Object is a fake object that’s used in place of a real object for the purpose of listening to the methods called on the mock object. The main job of a Mock Object is to ensure that the right methods get called on it._

**The difference between mocks and stubs**

_Test Stubs help with inputs and Mock Objects help with outputs. A Test Stub is a fake thing you stick in there to trick your program into working properly under test. A Mock Object is a fake thing you stick in there to spy on your program in the cases where you’re not able to test something directly._

File: test/game_statistics_test.rb [hunk 1]
```diff
@@ -19,9 +20,9 @@ def setup
   end
 
   def test_highest_total_score
-    game1 = MockGame.new(nil, 4, 6)
-    game2 = MockGame.new(nil, 2, 2)
-    game3 = MockGame.new(nil, 0, 3)
+    game1 = mock(home_goals: 4, away_goals: 6)
+    game2 = mock(home_goals: 2, away_goals: 2)
+    game3 = mock(home_goals: 0, away_goals: 3)
```
Comment (hashmaster3k) [inline test/game_statistics_test.rb:25]: Great use of abstraction!

File: test/game_statistics_test.rb [hunk 2]
```diff
@@ -46,10 +47,10 @@ def test_lowest_total_score
   end
 
   def test_percentage_home_wins
-    game_team1 = MockGameTeam.new('home', 'LOSS')
-    game_team2 = MockGameTeam.new('home', 'WIN')
-    game_team3 = MockGameTeam.new('home', 'WIN')
-    game_team4 = MockGameTeam.new('away', 'WIN')
+    game_team1 = mock(hoa: 'home', result: 'LOSS')
```
Comment (dhalverson) [inline test/game_statistics_test.rb:50]: These are making a bit more sense, looks good for testing purposes!

**Review by hashmaster3k — APPROVED** [review]
Excellent!

**Review by dhalverson — APPROVED** [review]
Looks good, testing looks complete

**Pull request merged**